### PR TITLE
Make \OCP\Mail strict

### PR DESCRIPTION
**Pull request by rullzer**
* Fix typehints
* Made strict

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: lib/private/Mail/Message.php [hunk 1]
```diff
@@ -204,7 +205,7 @@ public function getBcc() {
 	 * @param $subject
 	 * @return $this
 	 */
-	public function setSubject($subject) {
+	public function setSubject($subject): IMessage {
```
Comment (juliushaertl) [inline lib/private/Mail/Message.php]: I guess `$subject` should always be a string.

File: lib/private/Mail/EMailTemplate.php [hunk 2]
```diff
@@ -453,7 +454,7 @@ public function addBodyText($text, $plainText = '') {
 	 *   if empty the $metaInfo is used, if false none will be used
 	 * @since 12.0.0
 	 */
-	public function addBodyListItem($text, $metaInfo = '', $icon = '', $plainText = '', $plainMetaInfo = '') {
+	public function addBodyListItem(string $text, string $metaInfo = '', string $icon = '', string $plainText = '', string $plainMetaInfo = '') {
```
Comment (nickvergessen) [inline lib/private/Mail/EMailTemplate.php]: `$plain*` can also be false here, sorry

File: lib/public/Mail/IEMailTemplate.php [hunk 3]
```diff
@@ -105,7 +106,7 @@ public function addBodyText($text, $plainText = '');
 	 *   if empty the $metaInfo is used, if false none will be used
 	 * @since 12.0.0
 	 */
-	public function addBodyListItem($text, $metaInfo = '', $icon = '', $plainText = '', $plainMetaInfo = '');
+	public function addBodyListItem(string $text, string $metaInfo = '', string $icon = '', string $plainText = '', string $plainMetaInfo = '');
```
Comment (nickvergessen) [inline lib/public/Mail/IEMailTemplate.php]: `$plain*` can also be false here, sorry

**Review by nickvergessen — DISMISSED** [review]

**Comment by rullzer** [comment]
Another round :D

**Review by ChristophWurst — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**